common/ontology.yaml

```diff
@@ -0,0 +1,32 @@
+datasets:
+- data_type_def: OntologyTable
+  doc: A table for identifying which objects in a file contain
+       values that correspond to ontology terms
+  dtype:
+   - name: id
+     dtype: uint
+     doc: the unique identifier in this table
+   - name: object_id
+     dtype: text
+     doc: the UUID for the object that uses this ontology term
+   - name: field
+     dtype: text
+     doc: the field from the object (specified by object_id) that uses this ontological term
+   - name: item
+     dtype: uint
+     doc: an index into the OntologyMap that contains the term
+- data_type_def: OntologyMap
+  doc: A table for mapping user terms (i.e. keys) to ontology terms
+  dtype:
+   - name: id
+     dtype: uint
+     doc: the unique identifier in this table
+   - name: key
+     dtype: text
+     doc: the user key that maps to the ontology term
+   - name: ontology
+     dtype: text
+     doc: the ontology that the term comes from
+   - name: uri
+     dtype: text
+     doc: the unique resource identifier for the ontology term
```

Comment:
Should these tables have 
```
dims:
- num_rows
shape: 
- null
```
to restrict the tables to be 1-dimensional?

Reply:
This is a compound type. I don’t think individual types can be more than 1D

Reply:
I'm confused. Initially I thought that if the shape is not specified, the dataset can have any shape, but that's wrong - it actually means the dataset is a scalar dataset. Is this meant to be a scalar dataset?

Reply:
I misunderstood your comment. you are correct on setting dims/shape